### PR TITLE
Added pre/post custom scripts

### PR DESCRIPTION
**Pull request by himred**
In order to enhance customization, it would be handy to add custom scripts fired on events
This PR add 4 events:

- /config/custom-pre-start.sh is fired just before transmission is started
- /config/custom-post-start.sh is fired just after transmission is started
- /config/custom-pre-stop.sh is fired just before transmission is stopped
- /config/custom-post-stop.sh is fired just after transmission is stopped

This will enable a lot of possibilities, using custom scripts:
- add custom iptables rules
- change the vpn server dynamicly
- any other idea I don't have yet ;)

Himred

### CODE REVIEW

**Comment by haugene** [comment]
Hi! First of all, thanks for contributing. I think this can be a handy feature indeed. Sorry for not getting back to you earlier.

Allright. So I'm positive to enabling customization but I have a few comments:
- I propose that we put these kinds of scripts separate from /config. Maybe /scripts? The config directory is largely generated and contains sensitive information. I'm thinking this is something different and deserves its own space.
- Rename script paths from custom-pre-start.sh to transmission-pre-start.sh and so on. You mount it yourself so it's obviously custom, and in the future we might extend this to openvpn-pre-start.sh and proxy-pre-start.sh etc for the different services that might appear.
- Add a section in the readme describing the feature, usages and how to mount the needed files (remind people to make them executable and stuff)

What do you think?

**Comment by himred** [comment]
Hi! Thanks for your answer and acception my PR.
I agree with you about /scripts, this would be more logical than /config.
The user will be free to mount it or not if he needs the feature.

Again, I agree with the file naming, transmission-pre-start.sh is more logical and opens new incoming scripts if need.

Adding a section in the readme makes sense ;)

As I'm new to github, what is the right next step ? Should I send another PR ? modify this one (if possible) ?

Best regards,

Salim

**Comment by haugene** [comment]
Sweet. This PR is between your dev branch and the project dev branch. So you can just commit and push more changes to your dev branch and this PR will be updated automatically as it will always be the difference between the two branches.

**Comment by himred** [comment]
Ok, Thanks ! I'm on it.

File: README.md [hunk 1]
```diff
@@ -148,6 +148,36 @@ You may set the following parameters to customize the user id that runs transmis
 |`PUID` | Sets the user id who will run transmission | `PUID=1003`|
 |`PGID` | Sets the group id for the transmission user | `PGID=1003` |
 
+If you ever need to run custom code before or after transmission is executed or stopped, you can use the custom scripts feature.
+Custom scripts are located in the /scripts directory which is empty by default.
+To enable this feature, you'll need to mount the /scripts directory, ie:
+
+```
+$ docker run --cap-add=NET_ADMIN --device=/dev/net/tun -d \
```
Comment (haugene) [inline README.md]: Don't need this section. There's already a run statement and it's only a new mount.

File: README.md [hunk 2]
```diff
@@ -148,6 +148,36 @@ You may set the following parameters to customize the user id that runs transmis
 |`PUID` | Sets the user id who will run transmission | `PUID=1003`|
 |`PGID` | Sets the group id for the transmission user | `PGID=1003` |
 
+If you ever need to run custom code before or after transmission is executed or stopped, you can use the custom scripts feature.
```
Comment (haugene) [inline README.md]: Add a header?

**Comment by haugene** [comment]
Looking good! Just added a couple of more comments for the readme-section I hope you can update. And you need to merge or rebase against my dev branch and resolve the conflicts before I can merge.

**Comment by himred** [comment]
I had to redo a new PR to resolve conficts, I have posted it as PR #421 
I hope this was the right way.

**Comment by haugene** [comment]
merged #421 which supersedes

**Pull request closed without merging**